server/src/main/java/de/ingrid/igeserver/model/KeywordFilter.kt

```diff
@@ -0,0 +1,25 @@
+/**
+ * ==================================================
+ * Copyright (C) 2024 wemove digital solutions GmbH
+ * ==================================================
+ * Licensed under the EUPL, Version 1.2 or – as soon they will be
+ * approved by the European Commission - subsequent versions of the
+ * EUPL (the "Licence");
+ *
+ * You may not use this work except in compliance with the Licence.
+ * You may obtain a copy of the Licence at:
+ *
+ * https://joinup.ec.europa.eu/software/page/eupl
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the Licence is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the Licence for the specific language governing permissions and
+ * limitations under the Licence.
+ */
+package de.ingrid.igeserver.model
+
+abstract class KeywordFilter: QuickFilter() {
```

Comment:
Filter umbenennen. Es ist nicht nur ein allgemeiner Keyword-Filter, sondern ein spezieller Filter der für die OGC-API verwendet wird und eigentlich nur da. Deshalb hier besser "OgcApiSearchFilter".

Dort kann dann auch die gesamte Behandlung der SQL-Erstellung erfolgen, was bisher noch im Service in der Funktion "buildRecordsQuery" geschieht. Dabei sollten die gemeinsamen Teile hier in der abstrakten Klasse erstellt werden und in den Profilen nur die zusätzlichen SQL-Statements, die auch wirklich profilspezifisch sind. Wenn kein Profil angegeben ist, dann kann man ja einen DefaultFilter erstellen, der als Fallback genommen wird.

Reply:
Die Funktion "buildRecordsQuery" wurde aufgelöst und in eine Factory Struktur überführt. Die Funktionalität, um eine OGC spezifische ResearchQuery zu erstellen, wurden nach "OgcApiResearchQuery" ausgelagert. 

Für die einzelnen Profile gibt es Erweiterungen, um profilspezifische SQL Queries zu berücksichtigen:
* OgcApiResearchQueryIngrid
* OgcApiResearchQueryUvp
* OgcApiResearchQueryBmi

Die Benennung von "OgcApiResearchQuery" statt "OgcApiSearchFilter" ist erfolgt, da die Erstellung einer ResearchQuery für allen Profilen gleich ist und inkludiert werden konnte. 

DefaultFilter bzw. DefaultQuery wurde noch nicht aufgesetzt, da in dem Kontext der OGC Schnittstelle immer ein Katalog angegeben werden muss.

Reply:
Momentan liegen die profilspezifische SQL Research Queries unter `igeserver/profiles/*profileName*/quickFilter`.

Soll ich in dem Ordner `igeserver/ogc` einen Ordner `profiles` erstellen und dort profile-spezifische Funktionen sammeln?